components/overflow-group/overflow-group-mixin.js

```diff
@@ -0,0 +1,347 @@
+				itemLayout.trigger = 'force-show';
+				itemLayout.isChomped = false;
+				continue;
+			}
+
+			// handle maximum items to show
+			if (this.maxToShow >= 0 && showing.count >= this.maxToShow) {
+				isForcedOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'force-hide';
+				continue;
+			}
+
+			// chomp or unchomp based on space available, and we've already handled min/max above
+			if (!isSoftOverflowing && showing.width + itemLayout.width < this._availableWidth) {
+				showing.width += itemLayout.width;
+				showing.count += 1;
+				itemLayout.isChomped = false;
+				itemLayout.trigger = 'soft-show';
+
+			} else {
+				isSoftOverflowing = true;
+				itemLayout.isChomped = true;
+				itemLayout.trigger = 'soft-hide';
+
+			}
+
+		}
+		// if there is at least one showing and no more to be hidden, enable collapsing more button to [...]
+		this._overflowContainerHidden = this._itemLayouts.length === showing.count;
+		if (!this._overflowContainerHidden && (isSoftOverflowing || isForcedOverflowing)) {
+			for (let j = this._itemLayouts.length; j--;) {
+				if (showing.width + this._overflowContainerWidth < this._availableWidth) {
+					break;
+				}
+				const itemLayoutOverflowing = this._itemLayouts[j];
+				if (itemLayoutOverflowing.trigger !== 'soft-show') {
+					continue;
+				}
+				showing.width -= itemLayoutOverflowing.width;
+				showing.count -= 1;
+				isSoftOverflowing = true;
+				itemLayoutOverflowing.trigger = 'soft-hide';
+				itemLayoutOverflowing.isChomped = true;
+			}
+		}
+		const overflowDropdownOverflowing = (showing.width + this._overflowContainerWidth >= this._availableWidth);
+		const swapToMiniButton = overflowDropdownOverflowing && !this._overflowContainerHidden;
+
+		this.mini = this.openerType === OPENER_TYPE.ICON || swapToMiniButton;
+		this.chompIndex = this._overflowContainerHidden ? null : showing.count;
+
+		/** Dispatched when there is an update performed to the overflow group */
+		this.dispatchEvent(new CustomEvent('d2l-overflow-group-updated', { composed: false, bubbles: true }));
+	}
+
+	_getItemLayouts(filteredNodes) {
+		const items = filteredNodes.map((node) => {
+			const computedStyles = window.getComputedStyle(node);
+
+			return {
+				type: node.tagName.toLowerCase(),
+				isChomped: false,
+				isHidden: computedStyles.display === 'none',
+				width: Math.ceil(parseFloat(computedStyles.width) || 0)
+					+ parseInt(computedStyles.marginRight) || 0
+					+ parseInt(computedStyles.marginLeft) || 0,
+				node: node
+			};
+		});
+
+		return items.filter(({ isHidden }) => !isHidden);
+	}
+
+	async _getItems() {
+		// get the items from the button slot
+		this._slotItems = await this._getSlotItems();
+		// convert them to layout items (calculate widths)
+		this._itemLayouts = this._getItemLayouts(this._slotItems);
+		// convert to dropdown items (for overflow container)
+		this.dropdownItems = this._slotItems.map((node) => this.convertToOverflowItem(node));
```

Comment:
Do you think we should rename this? Is there anything left in the mixin that assumes the subclass will be handling the overflow with a dropdown?  Maybe we want to flip the naming around so this is more generic, and the variable that's used in the subclass (`d2l-overflow-group`) is more type specific?  Thoughts?

Reply:
I guess there are the class constants above too. 🤔

Reply:
Yup I think renaming this is a good idea. I think in general it would be some sort of dropdown but perhaps the mixin shouldn't make that assumption.

Reply:
Sounds good. If the mixin doesn't have any technical requirement for it to be a dropdown, then I think it makes sense. Maybe review for other dropdown references and use similar naming needed.